src/tvheadend/utilities/TCPSocket.h

```diff
@@ -0,0 +1,120 @@
+/*
+ *  Copyright (C) 2005-2020 Team Kodi
+ *  https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSE.md for more information.
+ */
+
+#pragma once
+
+#include "asio.hpp"
+
+#include "Logger.h"
+
+#include <chrono>
+
+namespace tvheadend
+{
+namespace utilities
+{
+
+class TCPSocket
+{
+public:
+  TCPSocket() = delete;
+  TCPSocket(const std::string& host, uint16_t port) : m_host(host), m_port(port) {}
+
+  virtual ~TCPSocket() { Close(); }
+
+  bool Open(uint64_t iTimeoutMs)
+  {
+    auto endpoints = asio::ip::tcp::resolver(m_context).resolve(m_host, std::to_string(m_port));
+
+    std::error_code error;
+    asio::async_connect(
+        m_socket, endpoints,
+        [&](const std::error_code& result_error, const asio::ip::tcp::endpoint&) {
+          error = result_error;
+        });
+
+    if (error)
```

Comment:
About here, as I understand his code of ASIO correct is the error not set until his work was done where is in independent thread, the `RunFor()` then only blocks until he is finished or timeout.

Reply:
Sorry, I do not understand what you mean.

Reply:
Fixed.